Automattic/wp-calypso
# Generated design picker: Use iframes for large previews

#### Changes proposed in this Pull Request

This PR replaces mShots with iframes for large design previews. As part of the design spec, we want to show the full design by scrolling the page, and not the iframe itself. To accomplish this, we added a message event handler which receives the page dimensions from the embedded site and resize the iframe's height using this payload.

Another thing we set out to do is to try to load as much of the iframes as early as possible, so that users can switch designs without needing to wait for the iframe to load each time. As an optimization, we load every iframe on page initialization and prioritize loading on the iframe users will see first.

Figma file for reference: 42PgVsdlQ1v2wpk6lmmrnq-fi-1601%3A59509
Loom recording for reference: https://www.loom.com/share/d2353287b0c646cf9d028796cf13d195 

CAVEAT: Currently there are a lot of designs, which means a lot of iframes loading in parallel. Once we only have three designs, I expect the iframe load time to improve.

![Screen Shot 2022-05-19 at 5 04 28 PM](https://user-images.githubusercontent.com/797888/169256652-61d15c0a-3f96-4a42-99ae-c02689136bfe.png)

#### Testing instructions



* Head to `/setup/vertical?siteSlug=${site_slug}` and make sure your site has a vertical set (I recommend Arts & Entertainment)
* Head to `/setup/intent?siteSlug=${site_slug}` and click on the "Start building" button.
* You should now land on the generated design picker screen.
* The design previews should now load in iframes.
  * Before done loading, they will show a loading message. 
  * After done loading, they will resize to the design's height.
* Also test that it works as per design in the mobile view.
* Also test switching to the traditional design picker and confirm it's working as before.
